app/src/main/java/com/example/superimagesviewer/viewmodel/MosaicViewModel.kt

```diff
@@ -4,17 +4,27 @@ import android.app.Application
 import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.LiveData
 import android.graphics.drawable.Drawable
+import com.example.superimagesviewer.Settings
 import com.example.superimagesviewer.repository.MosaicRepository
 
 class MosaicViewModel(application: Application) : AndroidViewModel(application) {
 
-    val userName: LiveData<String>
     val mosaicsList: LiveData<List<Drawable>>
     private val mosaicRepository: MosaicRepository = MosaicRepository(application)
 
     init {
         mosaicsList = mosaicRepository.getMosaicsList()
-        userName = mosaicRepository.getUserName()
+        refreshReferences(application)
+    }
+
+    private fun refreshReferences(application: Application) {
```

Comment:
Please rename to something more specific

Reply:
done